lib/src/domain/usecase/implementation/usecase_movie.dart

```diff
@@ -0,0 +1,61 @@
+import 'package:connectivity_plus/connectivity_plus.dart';
+
+import '../../../core/state/state.dart';
+import '../../../core/util/enum_category.dart';
+import '../../entity/movie.dart';
+import '../../repository/i_database_repository.dart';
+import '../../repository/i_repository.dart';
+import '../usecase_interface.dart';
+
+class MovieUseCase implements IUseCase {
+  MovieUseCase({
+    required this.repository,
+    required this.connectivity,
+    required this.movieDatabaseRepository,
+  });
+
+  IMovieRepository repository;
+  Connectivity connectivity;
+  IMovieDatabaseRepository movieDatabaseRepository;
+
+  @override
+  Future<DataState<List<Movie>>> repositoryCall(
+      [CategoryEnum? categoryCall]) async {
+    final ConnectivityResult connectivityResult =
+        await (connectivity.checkConnectivity());
+    late DataState<List<Movie>> dataStateMovie;
+    late List<Movie>? movies;
+
+    if (connectivityResult == ConnectivityResult.none) {
+      movies = await movieDatabaseRepository.getMovies(categoryCall!.category);
+      if (movies!.isEmpty) {
+        return DataState(
+          data: [],
+          state: DataEvents.empty,
+        );
+      } else {
+        return DataState(
+          data: movies,
+          state: DataEvents.success,
+        );
+      }
```

Comment:
Here you can made only on DataState combining both results like
```
return DataState(
          data: movies!.isEmpty ? [] : movies,
          state: movies!.isEmpty ? DataEvents.empty : DataEvents.empty,
);
```

Reply:
Done!